velox/functions/sparksql/SparkPartitionId.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/SparkPartitionId.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+class SparkPartitionId final : public exec::VectorFunction {
```

Comment:
Can this be implemented as a simple function?

Reply:
I implement this fucntion by `VectorFunction` because it can return constant vector as result, we don't need to set the partition id for each row. Please let me know if this makes sense to you.
Thanks.

Reply:
I expect this function to be constant folded, hence, it won't be invoked more than once. Therefore, a simple function should work just fine.

Reply:
Updated, thanks.